components/ThreeDotsMenü/ThreeDotMenü.js

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from "react";
+import Link from "next/link";
+import FavoriteButton from "../FavoriteButton/FavoriteButton";
+import useLocalStorageState from "use-local-storage-state";
+import { StyledLink, StyledUL, StyledLI, StyledButton } from "./styles";
+
+function ThreeDotMenu({ article }) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [favorites, setFavorites] = useLocalStorageState("favorites", []);
+  const [isFavorite, setIsFavorite] = useState(false);
+
+  useEffect(() => {
+    if (!favorites) return;
+    const found = favorites.some(
+      (favorite) => favorite.shareURL === article.shareURL
+    );
+    setIsFavorite(found);
+  }, [favorites, article.shareURL]);
+
+  function handleMarkFavorite() {
+    const currentFavorites = favorites || [];
+    let newFavorites = [...currentFavorites];
+    {
+      if (isFavorite) {
+        newFavorites = newFavorites.filter(
+          (favorite) => favorite.shareURL !== article.shareURL
+        );
+      } else {
+        newFavorites.push(article);
+      }
+      setFavorites(newFavorites);
+      setIsFavorite(!isFavorite);
+    }
+  }
```

Comment:
You don't need the outer braces.
```suggestion
      if (isFavorite) {
        newFavorites = newFavorites.filter(
          (favorite) => favorite.shareURL !== article.shareURL
        );
      } else {
        newFavorites.push(article);
      }
      setFavorites(newFavorites);
      setIsFavorite(!isFavorite);
    }
```